From 9b9810bf44830e5efec8071ced374c91590a6d90 Mon Sep 17 00:00:00 2001
From: Sugar Zhang <sugar.zhang@rock-chips.com>
Date: Mon, 13 Jun 2016 18:35:51 +0800
Subject: [PATCH] ARM64: dts: rk3399-evb: disable spdif sound default

this patch disable spdif sound in rk3399-evb.dtsi, and enable on
product dts side as required.

Change-Id: Icf61f13aeafdfae4c7a52603ce23cf2c04c27ab4
Signed-off-by: Sugar Zhang <sugar.zhang@rock-chips.com>
---
 .../boot/dts/rockchip/rk3399-evb-rev1-android.dts    | 12 ++++++++++++
 .../boot/dts/rockchip/rk3399-evb-rev2-android.dts    | 12 ++++++++++++
 arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi         |  5 +++--
 3 files changed, 27 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev1-android.dts b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev1-android.dts
index 8a22c5eef383..5cfff39b4f30 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev1-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev1-android.dts
@@ -113,3 +113,15 @@
 &hdmi_sound {
 	status = "okay";
 };
+
+&spdif {
+	status = "okay";
+};
+
+&spdif_out {
+	status = "okay";
+};
+
+&spdif_sound {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev2-android.dts b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev2-android.dts
index 46d040a07163..e08ec2769d63 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev2-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev2-android.dts
@@ -108,3 +108,15 @@
 &hdmi_sound {
 	status = "okay";
 };
+
+&spdif {
+	status = "okay";
+};
+
+&spdif_out {
+	status = "okay";
+};
+
+&spdif_sound {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi
index 6a7c8fa78ef6..210f23648833 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi
@@ -157,7 +157,8 @@
 		};
 	};
 
-	spdif-sound {
+	spdif_sound: spdif-sound {
+		status = "disabled";
 		compatible = "simple-audio-card";
 		simple-audio-card,name = "rockchip,spdif";
 		simple-audio-card,cpu {
@@ -169,6 +170,7 @@
 	};
 
 	spdif_out: spdif-out {
+		status = "disabled";
 		compatible = "linux,spdif-dit";
 		#sound-dai-cells = <0>;
 	};
@@ -272,7 +274,6 @@
 };
 
 &spdif {
-	status = "okay";
 	#sound-dai-cells = <0>;
 };
 
-- 
2.35.3

